scripts/components/cart.jsx

```diff
@@ -0,0 +1,504 @@
+			snCart.changeQuantity(item, qty, (newQty) => {
+				this.setState({ isLastStockKey: item.key });
+				callback(newQty);
+			});
+		});
+	}
+
+	onChangeVariant = (item, newVariantId) => {
+		if (item.id !== newVariantId) {
+			snCart.replaceItem(item.key, newVariantId, item.quantity);
+		}
+	}
+
+	onRemoveItem = (item) => {
+		snCart.removeItem(item.id);
+	}
+
+	onAddUpsell = (upsell) => {
+		if (upsell.upgrade_bundle_method === 'replace') {
+			snCart.replaceItem(upsell.targetId, upsell.replaceToId, 1);
+		} else {
+			snCart.addItem(upsell.replaceToId, 1);
+		}
+	}
+
+	onApplyDiscountCode = (code) => {
+		this.setState({ loadingDiscount: true }, () => {
+			snCart.applyDiscountCode(code).then((discountData) => {
+				if (discountData.enabled === false || discountData.isValid === false) {
+					this.setState({
+						loadingDiscount: false,
+						discountData: this.getDiscountDataDisplay({ reason: discountData.error }),
+					});
+				} else {
+					this.setState({
+						loadingDiscount: false,
+						discountData: this.getDiscountDataDisplay({
+							...discountData,
+							valid: true,
+							error: '',
+						}),
+					});
+				}
+			});
+		});
+	}
+
+	onRemoveDiscountCode = () => {
+		this.setState({ loadingDiscount: true }, () => {
+			snCart.removeDiscountCode();
+			this.setState({
+				loadingDiscount: false,
+				discountData: this.getDiscountDataDisplay(),
+			});
+		});
+	}
+
+	onToggleManualGwp = (id) => {
+		this.setState({
+			loadingManualGwp: { loading: true, id },
+		}, () => {
+			const newSelected = snCart.toggleManualGwp(this.state.manualGwp, id);
+			this.setState((prevState) => ({
+				manualGwp: {
+					...prevState.manualGwp,
+					selectedKey: newSelected,
+				},
+				loadingManualGwp: { loading: false, id: -1 },
+			}));
+		});
+	}
+
+	handleKeyDown = (e) => {
+		if (e.keyCode === 13) {
+			e.preventDefault();
+		}
+	}
+
+	submitForm() {
+		if (this.formRef) {
+			this.formRef.submit();
+		}
+	}
+
+	render() {
+		const {
+			cart,
+			loadingInit,
+			isLastStockKey,
+			itemCount,
+			totalPrice,
+			loadingDiscount,
+			discountData,
+			manualGwp,
+			loadingManualGwp,
+			upsellData,
+			shippingData,
+			discountMeter,
+		} = this.state;
+		return (
+			<div className="modal-dialog modal-md m-0 w-100 float-right">
+				<div className="modal-content vh-100 border-0 rounded-0">
+					<div className="modal-body mobile-wrapper pt-0 px-lg-0">
+						<div className="container d-flex flex-column align-items-stretch text-center pt-2">
+							<h2>{tStrings.cart_drawer_title}</h2>
+							<button type="button" className="close m-0 p-1 position-absolute" data-dismiss="modal" aria-label="Close">
+								<span className="sni sni__close" aria-hidden="true" />
+							</button>
+
+							{tSettings.cartShippingMeter.enable && discountMeter.enabled && (
+								<CartDiscountMeter
+									target={discountMeter.target}
+									current={discountMeter.current}
+								/>
+							)}
+							<hr className="w-100 m-0" />
```

Comment:
please give a top margin to this element, since it's attached to the progress. Or maybe better bottom-margin to the progress